### PR TITLE
Opgops 2005 ansible23

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: roles/ec2-app/tasks/destroy_elasticache.yml [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+  set_fact:
+    ecache_id_len: "{{  [ec_data.ec_name, opg_data.stack ] | join('-') | length  }}"
+
 - name: generate cluster identifier
   set_fact:
     ecc_instance_name: "{{ [ec_data.ec_name, opg_data.stack] | join('-') }}"
-  when: "{{ [ec_data.ec_name, opg_data.stack ] | join('-') | length <= 20}}"
+  when: ecache_id_len <= 20
 
 - name: generate cluster identifier
```
Comment (carinadigital) [inline roles/ec2-app/tasks/destroy_elasticache.yml]: Same name for both options?

File: roles/ec2-app/tasks/destroy_stack.yml [hunk 2]
```diff
@@ -18,9 +18,15 @@
...
   register: launch_configs
 
+- name: Get list of autoscaling groups
+  ec2_asg_facts:
+    tags:
+      Environment: "{{ opg_data.environment }}"
+      Stack: "{{ target }}"
+  register: asg_list
+
 - name: get list of autoscaling groups
```
Comment (carinadigital) [inline roles/ec2-app/tasks/destroy_stack.yml]: name ->` get instance list` or something similar?

File: roles/ec2-app/tasks/destroy_elasticache.yml (hunk 1 above)
Comment (brettminnie) [reply roles/ec2-app/tasks/destroy_elasticache.yml]: Well yes, its basically an if else

Comment (carinadigital) [reply roles/ec2-app/tasks/destroy_elasticache.yml]: I always find it difficult reading ansible output and what was done, so I was thinking  `generate cluster identifier long ecache` and `generate cluster identifier short ecache`

Comment (brettminnie) [reply roles/ec2-app/tasks/destroy_elasticache.yml]: Right, task name, be more specific :smile:

File: roles/ec2-app/tasks/elasticache.yml [hunk 3]
```diff
@@ -42,15 +42,19 @@
...
+  set_fact:
+    ecache_id_len: "{{  [ec_data.ec_name, opg_data.stack ] | join('-') | length  }}"
+
 - name: generate cluster identifier
   set_fact:
     ecc_instance_name: "{{ [ec_data.ec_name, opg_data.stack] | join('-') }}"
-  when: "{{ [ec_data.ec_name, opg_data.stack ] | join('-') | length <= 20}}"
+  when: ecache_id_len <= 20
 
 - name: generate cluster identifier
```
Comment (carinadigital) [inline roles/ec2-app/tasks/elasticache.yml]: Same naming for these 2 tasks as above to be used here? e.g. `generate truncated cluster identifier` `

**Review by carinadigital — APPROVED** [review]

**Review by tim-mcarthur-moj — APPROVED** [review]

**Pull request merged**